pyat/at/load/matfile.py

```diff
@@ -21,7 +25,103 @@ def _load_element(index, element_array, check=True, quiet=False):
     return element_from_dict(kwargs, index=index, check=check, quiet=quiet)
 
 
-def load_mat(filename, key=None, check=True, quiet=False):
+def _scanner(elems, **kwargs):
```

Comment:
Now this isn't a generator, it seems to me that it could be used whenever making a lattice from a list and so could be public?

Reply:
Since the `RingParam` element does not exist any more, a large part of this scan is now useless. Do the remaining tests have any interest for a list created in another way than a Matlab file? If someone is building a lattice by hand, he should be able to specify `energy` and `periodicity` to the `Lattice` constructor. I recall that `energy` is not provided to any python element constructor (unless you put it in a keyword argument). Again, no more scan at `Lattice` creation removes the overhead. Since any subscripting now creates a new Lattice, this has some interest.
For the time being, `load_mat` may still change as I am looking at importing other formats, and I would like to reuse as much code as possible. Should a scan of `list`s for energy or periodicity still be useful, I would wait a little bit.